pkg/net/libp2p/transport_test.go

```diff
@@ -770,3 +825,121 @@ func TestLibp2p_OpeningConnectionAfterFail(t *testing.T) {
 	m.CloseAllHosts()
 	m.testNoHostConnections()
 }
+
+// TestMessagingWithNewNodes tests that the transport operates normally if new nodes are connected.
+//
+// nolint:gocognit
+func TestMessagingWithNewNodes(t *testing.T) {
+	logger := logging.ConsoleDebugLogger
+
+	N := 10 // number of nodes
+	M := 5  // number of new nodes
+	var nodes []types.NodeID
+	for i := 0; i < N+M; i++ {
+		nodes = append(nodes, types.NodeID(fmt.Sprint(i)))
+	}
+
+	m := newMockLibp2pCommunication(t, DefaultParams(), nodes, logger)
+	m.StartAllTransports()
+	initialNodes := m.Membership(nodes[:N]...)
+	allNodes := m.Membership(nodes...)
+
+	t.Logf(">>> connecting nodes")
+	for i := 0; i < N; i++ {
+		m.transports[nodes[i]].Connect(initialNodes)
+	}
+
+	for i := 0; i < N; i++ {
+		for j := 0; j < N; j++ {
+			if i == j {
+				continue
+			}
+			m.testEventuallyConnected(nodes[i], nodes[j])
+		}
+	}
+
+	t.Log(">>> sending messages")
+
+	var received, sent int64
+	var senders, receivers sync.WaitGroup
+
+	sender := func(src, dst types.NodeID) {
+		defer senders.Done()
+
+		for i := 0; i < 100; i++ {
+			time.Sleep(time.Duration(rand.Intn(200)) * time.Millisecond) // nolint
+			err := m.transports[src].Send(dst, &messagepb.Message{DestModule: "iss", Type: &messagepb.Message_Iss{}})
+			if err != nil {
+				t.Logf("%v->%v failed to send: %v", src, dst, err)
+			}
+			atomic.AddInt64(&sent, 1)
```

Comment:
Should we increment the counter even if the sending failed?

Reply:
Fixed.